app/components/NewChannelItem/index.jsx

```diff
@@ -0,0 +1,68 @@
+import React, {PropTypes} from 'react';
+import {Map} from 'immutable';
+import './styles.scss';
+
+export default class ChannelItem extends React.Component {
+
+  static propTypes = {
+    replaceDirtyChannel: PropTypes.func.isRequired,
+    newChannel: PropTypes.func.isRequired,
+  }
+
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      channelName: '',
+    };
+  }
+
+
+  componentDidMount = () => {
+    const newChannelItem = this.refs.newChannel.getDOMNode();
+    const newChannelInput = this.refs.newChannelInput.getDOMNode();
+    setTimeout(function timeout() {
+      newChannelItem.className = newChannelItem.className + ' new-channel_animated';
+      newChannelInput.focus();
+    }, 1);
```

Comment:
And we can use react-motion to do animation easy (use willEnter and willLeave props)